src/components/PostDetails.tsx

```diff
@@ -1,118 +1,190 @@
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  selectedPost: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost: {
+    id,
+    title,
+    body,
+  },
+}) => {
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [isButtonHidden, setIsButtonHidden] = useState(false);
+  const [isRemoveError, setIsRemoveError] = useState(false);
+
+  const getPostsComments = async () => {
+    try {
+      setIsLoading(true);
+      setIsLoadingError(false);
+      setComments(null);
+      setIsButtonHidden(false);
+
+      const commentsFromServer = await getComments(id);
+
+      setComments(commentsFromServer);
+      setIsLoading(false);
+    } catch (error) {
+      setIsLoadingError(true);
+      setIsLoading(false);
+    }
+  };
+
+  const onButtonClick = () => {
+    setIsButtonHidden(true);
+  };
+
+  const handleCommentAddition = (newComment: Comment) => {
+    setComments(prevState => {
+      if (prevState) {
+        return [
+          ...prevState,
+          newComment,
+        ];
+      }
+
+      return [newComment];
+    });
+  };
+
+  const removeComment = (commentId: number) => {
+    setComments(prevState => {
+      if (prevState) {
+        return prevState.filter(comment => comment.id !== commentId);
+      }
+
+      return null;
+    });
+  };
+
+  const handleCommentRemove = async (commentId: number) => {
+    try {
+      removeComment(commentId);
+
+      await deleteComment(commentId);
+    } catch (error) {
+      setIsRemoveError(true);
+
+      setTimeout(() => {
+        setIsRemoveError(false);
+      }, 3000);
+    }
+  };
+
+  useEffect(() => {
+    getPostsComments();
+  }, [id]);
+
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${id}: ${title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
+          {isLoading && <Loader />}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+          {(isLoadingError) && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
+            </div>
+          )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {comments && (
+            comments.length === 0 ? (
```

Comment:
Same here

Reply:
Changed checks for conditional rendering